org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/DetectVMInstallationsJob.java

```diff
@@ -48,12 +48,17 @@ public class DetectVMInstallationsJob extends Job {
 
 	private static final Object FAMILY = DetectVMInstallationsJob.class;
 
+	private static final String CI_ENV_VARIABLE = System.getProperty("jdt.jvmdetect.envCI", "CI"); //$NON-NLS-1$ //$NON-NLS-2$
+
 	public DetectVMInstallationsJob() {
 		super(LaunchingMessages.lookupInstalledJVMs);
 	}
 
 	@Override
 	public IStatus run(IProgressMonitor monitor) {
+		if (!CI_ENV_VARIABLE.isEmpty() && Boolean.parseBoolean(System.getenv(CI_ENV_VARIABLE))) {
+			return Status.OK_STATUS;
+		}
```

Comment:
Enablement logic should be placed with other similar code, in `initialize`.

Reply:
valid request + org.eclipse.jdt.internal.launching.DetectVMInstallationsJob.initialize() already has a flag "DetectVMInstallationsJob.disabled" which can be used for this purpose. On top of that there is a setting org.eclipse.jdt.internal.launching.LaunchingPlugin.PREF_DETECT_VMS_AT_STARTUP, i don't see a value in introducing even a third option

Reply:
The main point is that this is something everyone needs to know and then needs to set ... so one probably wastes a lot of time trying to find out why Tycho build is "unstable"... then probably one is lucky enough to map it to this feature and set the option hopefully not forget it next time again.

But yes initialize is probably a better place.